src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPLY_TEXT = "supply";
+    private static final String BUY_TEXT = "buy";
+    private static final String RESULT_TEXT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readFromFile(fromFileName);
+        String[] calculated = calculate(data);
+        writeToFile(toFileName, calculated);
+    }
+
+    private String[] readFromFile(String fileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line = reader.readLine();
+            while (line != null) {
+                stringBuilder.append(line).append(System.lineSeparator());
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fileName, e);
+        }
+        return stringBuilder.toString().split(System.lineSeparator());
+    }
+
+    private void writeToFile(String fileName, String[] calculate) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(fileName))) {
+            for (String data : calculate) {
+                bufferedWriter.write(data);
+                bufferedWriter.write(System.lineSeparator());
+            }
```

Comment:
Let's compose the result first and then write everything to the file in a one method call